docs/reference/configuration.md

```diff
@@ -28,15 +28,43 @@ This reference uses `.` to denote the nesting of values.
 * `logging.files_keep` `(int : 0)` - Number of log files to keep, default is all.
 * `actions.enabled` `(bool : true)` - Setting this to false will block hooks from being executed
-* `database.connection_string` `(string : "postgres://localhost:5432/postgres?sslmode=disable")` - PostgreSQL connection string to use
-* `database.max_open_connections` `(int : 25)` - Maximum number of open connections to the database
-* `database.max_idle_connections` `(int : 25)` - Sets the maximum number of connections in the idle connection pool
-* `database.connection_max_lifetime` `(duration : 5m)` - Sets the maximum amount of time a connection may be reused
+* ~~`database.connection_string` `(string : "postgres://localhost:5432/postgres?sslmode=disable")` - PostgreSQL connection string to use~~
+* ~~`database.max_open_connections` `(int : 25)` - Maximum number of open connections to the database~~
+* ~~`database.max_idle_connections` `(int : 25)` - Sets the maximum number of connections in the idle connection pool~~
+* ~~`database.connection_max_lifetime` `(duration : 5m)` - Sets the maximum amount of time a connection may be reused~~
+
+  **Note:** Deprecated - See `database` section 
+  {: .note }
+* `database` - Configuration section for the lakeFS key-value store database
+  + `database.type` `(string : ["postgres"|"dynamodb"])` - lakeFS database type 
+  + `database.postgres` - Configuration section when using `database.type="postgres"`
+    + `database.postgres.connection_string` `(string : "postgres://localhost:5432/postgres?sslmode=disable")` - PostgreSQL connection string to use
+    + `database.postgres.max_open_connections` `(int : 25)` - Maximum number of open connections to the database
+    + `database.postgres.max_idle_connections` `(int : 25)` - Sets the maximum number of connections in the idle connection pool
+    + `database.postgres.connection_max_lifetime` `(duration : 5m)` - Sets the maximum amount of time a connection may be reused
+  + `database.beta_dynamodb` - Configuration section when using `database.type="dynamodb"`
+    + `database.beta_dynamodb.table_name` `(string : kvstore)` -
+    + `database.beta_dynamodb.read_capacity_units` `(int : 1000)` - Throughput parameter for read capacity
+    + `database.beta_dynamodb.write_capacity_units` `(int : 1000)` - Throughput parameter for write capacity
+
+       **Note:** Refer to the following [AWS documentation](https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/Limits.html) for further information
+       {: .note }
+    + `database.beta_dynamodb.scan_limit` `(int : 0)` - Maximal number of items per page during scan operation
+    + `database.beta_dynamodb.endpoint` `(string : required)` - Endpoint URL for database instance
+    + `database.beta_dynamodb.aws_region` `(string : )` - AWS Region of database instance
+    + `database.beta_dynamodb.aws_access_key_id` `(string : )` - AWS access key ID
```

Comment:
The defaults here are wrong or confusing (I think).
The default endpoint is the AWS's DynamoDB endpoint.
Default credentials are the one granted to the server, just like with S3 blockstore. No differences there.

Reply:
For real production usecases, the users would probably need to leave these empty.

The reason we have them is for testing against a local-DynamoDB container which needs different creds.

Reply:
Is it not a possible usecase the the user will have the database on a different AWS account?